public/app/plugins/panel/graph/module.ts

```diff
@@ -156,7 +156,10 @@ class GraphCtrl extends MetricsPanelCtrl {
       panel: this.panel,
       range: this.range,
     });
-    return super.issueQueries(datasource);
+
+    return this.annotationsSrv.datasourcePromises.then(r => {
```

Comment:
maybe add a comment here that this allows annotations service a change to fire annotations queries before graph queries (fire not wait for completion).

Reply:
Commented and added a note about the original issue it fixes.